cloudevents/sdk/event/base.py

```diff
@@ -20,61 +20,141 @@
 # TODO(slinkydeveloper) is this really needed?
 class EventGetterSetter(object):
 
+    # ce-specversion
     def CloudEventVersion(self) -> str:
         raise Exception("not implemented")
 
-    # CloudEvent attribute getters
-    def EventType(self) -> str:
-        raise Exception("not implemented")
+    @property
+    def specversion(self):
+        return self.CloudEventVersion()
 
-    def Source(self) -> str:
+    def SetCloudEventVersion(self, specversion: str) -> object:
         raise Exception("not implemented")
 
-    def EventID(self) -> str:
-        raise Exception("not implemented")
+    @specversion.setter
+    def specversion(self, value: str):
+        self.SetCloudEventVersion(value)
 
-    def EventTime(self) -> str:
+    # ce-type
+    def EventType(self) -> str:
         raise Exception("not implemented")
 
-    def SchemaURL(self) -> str:
-        raise Exception("not implemented")
+    @property
+    def type(self):
+        return self.EventType()
 
-    def Data(self) -> object:
+    def SetEventType(self, eventType: str) -> object:
         raise Exception("not implemented")
 
-    def Extensions(self) -> dict:
-        raise Exception("not implemented")
+    @type.setter
+    def type(self, value: str):
+        self.SetEventType(value)
 
-    def ContentType(self) -> str:
+    # ce-source
+    def Source(self) -> str:
         raise Exception("not implemented")
 
-    # CloudEvent attribute constructors
-    # Each setter return an instance of its class
-    #      in order to build a pipeline of setter
-    def SetEventType(self, eventType: str) -> object:
-        raise Exception("not implemented")
+    @property
+    def source(self):
+        return self.Source()
 
     def SetSource(self, source: str) -> object:
         raise Exception("not implemented")
 
+    @source.setter
+    def source(self, value: str):
+        self.SetSource(value)
+
+    # ce-id
+    def EventID(self) -> str:
+        raise Exception("not implemented")
+
+    @property
+    def id(self):
+        return self.EventID()
+
     def SetEventID(self, eventID: str) -> object:
         raise Exception("not implemented")
 
+    @id.setter
+    def id(self, value: str):
+        self.SetEventID(value)
+
+    # ce-time
+    def EventTime(self) -> str:
+        raise Exception("not implemented")
+
+    @property
+    def time(self):
+        return self.EventTime()
+
     def SetEventTime(self, eventTime: str) -> object:
         raise Exception("not implemented")
```

Comment:
Do we still need these methods?

Reply:
Unfortunately users are using the old getters/setters. To maintain backwards compatibility I left some redundancy

Reply:
Unfortunately yes. Removing them would break existing user code